In [958]:
%%html
<style>
    body {
        font-family: "Times New Roman", Times, serif; 
    }
</style>

In [959]:
import pandas as pd
import numpy as np 
import seaborn as sns
import matplotlib.pyplot as plt

filename = "master.csv"
df = pd.read_csv(filename)

df.rename(columns={"HDI for year":"HDI_for_year",
                  " gdp_for_year ($) ":"gdp_for_year"," gdp_per_capita ($) ":"gdp_per_capita",
                    "gdp_per_capita ($)":"gdp_per_capita"}, inplace=True)

df["gdp_for_year"] = df["gdp_for_year"].str.replace(",","").astype(np.int64)
df["age"] = df["age"].str.replace("5-14 years","05-14 years")
df

country  year     sex          age  suicides_no  population  \
0         Albania  1987    male  15-24 years           21      312900   
1         Albania  1987    male  35-54 years           16      308000   
2         Albania  1987  female  15-24 years           14      289700   
3         Albania  1987    male    75+ years            1       21800   
4         Albania  1987    male  25-34 years            9      274300   
...           ...   ...     ...          ...          ...         ...   
27815  Uzbekistan  2014  female  35-54 years          107     3620833   
27816  Uzbekistan  2014  female    75+ years            9      348465   
27817  Uzbekistan  2014    male  05-14 years           60     2762158   
27818  Uzbekistan  2014  female  05-14 years           44     2631600   
27819  Uzbekistan  2014  female  55-74 years           21     1438935   

       suicides/100k pop    country-year  HDI_for_year  gdp_for_year  \
0                   6.71     Albania1987           NaN    2156624900   
1                   5.19     Albania1987           NaN    2156624900   
2                   4.83     Albania1987           NaN    2156624900   
3                   4.59     Albania1987           NaN    2156624900   
4                   3.28     Albania1987           NaN    2156624900   
...                  ...             ...           ...           ...   
27815               2.96  Uzbekistan2014         0.675   63067077179   
27816               2.58  Uzbekistan2014         0.675   63067077179   
27817               2.17  Uzbekistan2014         0.675   63067077179   
27818               1.67  Uzbekistan2014         0.675   63067077179   
27819               1.46  Uzbekistan2014         0.675   63067077179   

       gdp_per_capita       generation  
0                 796     Generation X  
1                 796           Silent  
2                 796     Generation X  
3                 796  G.I. Generation  
4                 796          Boomers  
...               ...              ...  
27815            2309     Generation X  
27816            2309           Silent  
27817            2309     Generation Z  
27818            2309     Generation Z  
27819            2309          Boomers  

[27820 rows x 12 columns]

In [960]:
df.isnull().sum()

country                  0
year                     0
sex                      0
age                      0
suicides_no              0
population               0
suicides/100k pop        0
country-year             0
HDI_for_year         19456
gdp_for_year             0
gdp_per_capita           0
generation               0
dtype: int64

In [961]:
unique_counts = df.nunique()

# Print the unique counts for each feature
print(unique_counts)

country                101
year                    32
sex                      2
age                      6
suicides_no           2084
population           25564
suicides/100k pop     5298
country-year          2321
HDI_for_year           305
gdp_for_year          2321
gdp_per_capita        2233
generation               6
dtype: int64


preprocess the data with previous methods.

In [962]:
df=df.drop(columns=['HDI_for_year', 'suicides_no', 'country-year','gdp_per_capita','population'])
df

country  year     sex          age  suicides/100k pop  gdp_for_year  \
0         Albania  1987    male  15-24 years               6.71    2156624900   
1         Albania  1987    male  35-54 years               5.19    2156624900   
2         Albania  1987  female  15-24 years               4.83    2156624900   
3         Albania  1987    male    75+ years               4.59    2156624900   
4         Albania  1987    male  25-34 years               3.28    2156624900   
...           ...   ...     ...          ...                ...           ...   
27815  Uzbekistan  2014  female  35-54 years               2.96   63067077179   
27816  Uzbekistan  2014  female    75+ years               2.58   63067077179   
27817  Uzbekistan  2014    male  05-14 years               2.17   63067077179   
27818  Uzbekistan  2014  female  05-14 years               1.67   63067077179   
27819  Uzbekistan  2014  female  55-74 years               1.46   63067077179   

            generation  
0         Generation X  
1               Silent  
2         Generation X  
3      G.I. Generation  
4              Boomers  
...                ...  
27815     Generation X  
27816           Silent  
27817     Generation Z  
27818     Generation Z  
27819          Boomers  

[27820 rows x 7 columns]

In [963]:
age_mode = df['age'].mode()[0]
df.fillna({'age': age_mode}, inplace=True)

df = pd.get_dummies(df, columns=['sex', 'age','country','generation'],drop_first=True)
df

year  suicides/100k pop  gdp_for_year  sex_male  age_15-24 years  \
0      1987               6.71    2156624900      True             True   
1      1987               5.19    2156624900      True            False   
2      1987               4.83    2156624900     False             True   
3      1987               4.59    2156624900      True            False   
4      1987               3.28    2156624900      True            False   
...     ...                ...           ...       ...              ...   
27815  2014               2.96   63067077179     False            False   
27816  2014               2.58   63067077179     False            False   
27817  2014               2.17   63067077179      True            False   
27818  2014               1.67   63067077179     False            False   
27819  2014               1.46   63067077179     False            False   

       age_25-34 years  age_35-54 years  age_55-74 years  age_75+ years  \
0                False            False            False          False   
1                False             True            False          False   
2                False            False            False          False   
3                False            False            False           True   
4                 True            False            False          False   
...                ...              ...              ...            ...   
27815            False             True            False          False   
27816            False            False            False           True   
27817            False            False            False          False   
27818            False            False            False          False   
27819            False            False             True          False   

       country_Antigua and Barbuda  ...  country_United Arab Emirates  \
0                            False  ...                         False   
1                            False  ...                         False   
2                            False  ...                         False   
3                            False  ...                         False   
4                            False  ...                         False   
...                            ...  ...                           ...   
27815                        False  ...                         False   
27816                        False  ...                         False   
27817                        False  ...                         False   
27818                        False  ...                         False   
27819                        False  ...                         False   

       country_United Kingdom  country_United States  country_Uruguay  \
0                       False                  False            False   
1                       False                  False            False   
2                       False                  False            False   
3                       False                  False            False   
4                       False                  False            False   
...                       ...                    ...              ...   
27815                   False                  False            False   
27816                   False                  False            False   
27817                   False                  False            False   
27818                   False                  False            False   
27819                   False                  False            False   

       country_Uzbekistan  generation_G.I. Generation  \
0                   False                       False   
1                   False                       False   
2                   False                       False   
3                   False                        True   
4                   False                       False   
...                   ...                         ...   
27815                True                       Fals

In [964]:
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report, accuracy_score

X = df.drop(columns=['suicides/100k pop'])
y = df['suicides/100k pop']
X

year  gdp_for_year  sex_male  age_15-24 years  age_25-34 years  \
0      1987    2156624900      True             True            False   
1      1987    2156624900      True            False            False   
2      1987    2156624900     False             True            False   
3      1987    2156624900      True            False            False   
4      1987    2156624900      True            False             True   
...     ...           ...       ...              ...              ...   
27815  2014   63067077179     False            False            False   
27816  2014   63067077179     False            False            False   
27817  2014   63067077179      True            False            False   
27818  2014   63067077179     False            False            False   
27819  2014   63067077179     False            False            False   

       age_35-54 years  age_55-74 years  age_75+ years  \
0                False            False          False   
1                 True            False          False   
2                False            False          False   
3                False            False           True   
4                False            False          False   
...                ...              ...            ...   
27815             True            False          False   
27816            False            False           True   
27817            False            False          False   
27818            False            False          False   
27819            False             True          False   

       country_Antigua and Barbuda  country_Argentina  ...  \
0                            False              False  ...   
1                            False              False  ...   
2                            False              False  ...   
3                            False              False  ...   
4                            False              False  ...   
...                            ...                ...  ...   
27815                        False              False  ...   
27816                        False              False  ...   
27817                        False              False  ...   
27818                        False              False  ...   
27819                        False              False  ...   

       country_United Arab Emirates  country_United Kingdom  \
0                             False                   False   
1                             False                   False   
2                             False                   False   
3                             False                   False   
4                             False                   False   
...                             ...                     ...   
27815                         False                   False   
27816                         False                   False   
27817                         False                   False   
27818                         False                   False   
27819                         False                   False   

       country_United States  country_Uruguay  country_Uzbekistan  \
0                      False            False               False   
1                      False            False               False   
2                      False            False               False   
3                      False            False               False   
4                      False            False               False   
...                      ...              ...                 ...   
27815                  False            False                True   
27816                  False            False                True   
27817                  False            False                True   
27818                  False            False                True   
27819                  False            False                True   

       generation_G.I. Generation  generation_Generation X  \
0                           False       

## 1. Use your previous pre-processed dataset, keep the variables as one-hot encoded, and develop a multiple linear regression model. How many regression coefficients does this model have?


Fit the Multiple Linear Regression Model

In [965]:
from sklearn.linear_model import LinearRegression

model = LinearRegression()
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42,shuffle=True)

model.fit(X_train, y_train)

num_coefficients = len(model.coef_) + 1 # Add the intercept

print("Number of regression coefficients:", num_coefficients)

Number of regression coefficients: 114


The number equals to the number of features (columns), plus intercept, which is 114.

## 2. Use this model to predict the target variable for people with age 20, male, and generation X. Report this prediction. What is the MAE error of this prediction?

In [966]:
target_data = df[(df['age_15-24 years'] == 1) & (df['sex_male'] == 1) & (df['generation_Generation X'] == 1)]

target_X = target_data.drop('suicides/100k pop', axis=1)
target_y = target_data['suicides/100k pop']
target_X

year  gdp_for_year  sex_male  age_15-24 years  age_25-34 years  \
0      1987    2156624900      True             True            False   
13     1988    2126000000      True             True            False   
28     1989    2335124988      True             True            False   
37     1992     709452584      True             True            False   
48     1993    1228071038      True             True            False   
...     ...           ...       ...              ...              ...   
27632  1996   13948892216      True             True            False   
27644  1997   14744603774      True             True            False   
27656  1998   14988971211      True             True            False   
27667  1999   17078465982      True             True            False   
27679  2000   13760374488      True             True            False   

       age_35-54 years  age_55-74 years  age_75+ years  \
0                False            False          False   
13               False            False          False   
28               False            False          False   
37               False            False          False   
48               False            False          False   
...                ...              ...            ...   
27632            False            False          False   
27644            False            False          False   
27656            False            False          False   
27667            False            False          False   
27679            False            False          False   

       country_Antigua and Barbuda  country_Argentina  ...  \
0                            False              False  ...   
13                           False              False  ...   
28                           False              False  ...   
37                           False              False  ...   
48                           False              False  ...   
...                            ...                ...  ...   
27632                        False              False  ...   
27644                        False              False  ...   
27656                        False              False  ...   
27667                        False              False  ...   
27679                        False              False  ...   

       country_United Arab Emirates  country_United Kingdom  \
0                             False                   False   
13                            False                   False   
28                            False                   False   
37                            False                   False   
48                            False                   False   
...                             ...                     ...   
27632                         False                   False   
27644                         False                   False   
27656                         False                   False   
27667                         False                   False   
27679                         False                   False   

       country_United States  country_Uruguay  country_Uzbekistan  \
0                      False            False               False   
13                     False            False               False   
28                     False            False               False   
37                     False            False               False   
48                     False            False               False   
...                      ...              ...                 ...   
27632                  False            False                True   
27644                  False            False                True   
27656                  False            False                True   
27667                  False            False                True   
27679                  False            False                True   

       generation_G.I. Generation  generation_Generation X  \
0                           False       

In [967]:
y_pred = model.predict(target_X)
print(f"Predicted suicide rate (suicides/100k pop): {np.mean(y_pred):.2f}")

Predicted suicide rate (suicides/100k pop): 17.29


In [968]:
from sklearn.metrics import mean_absolute_error

mae = mean_absolute_error(target_y, y_pred)
print("Mean Absolute Error (MAE):", mae)

Mean Absolute Error (MAE): 6.921515173141622


The model predicts a suicide rate of approximately 17.29 suicides per 100,000 people, which is high. According to the MAE, this model may not fully capture the model's performance.

## 3. Now go back to the original sex, age, and generation variables in their original numerical form. How many line coefficients are there?

In [969]:
filename = "master.csv"
df = pd.read_csv(filename)

df.rename(columns={"HDI for year":"HDI_for_year",
                  " gdp_for_year ($) ":"gdp_for_year"," gdp_per_capita ($) ":"gdp_per_capita",
                    "gdp_per_capita ($)":"gdp_per_capita"}, inplace=True)
df["gdp_for_year"] = df["gdp_for_year"].str.replace(",","").astype(np.int64)
df["age"] = df["age"].str.replace("5-14 years","05-14 years")

age_mode = df['age'].mode()[0]
df.fillna({'age': age_mode}, inplace=True)
most_common_country = df['country'].mode()[0]
most_recent_year = df['year'].max()
df

country  year     sex          age  suicides_no  population  \
0         Albania  1987    male  15-24 years           21      312900   
1         Albania  1987    male  35-54 years           16      308000   
2         Albania  1987  female  15-24 years           14      289700   
3         Albania  1987    male    75+ years            1       21800   
4         Albania  1987    male  25-34 years            9      274300   
...           ...   ...     ...          ...          ...         ...   
27815  Uzbekistan  2014  female  35-54 years          107     3620833   
27816  Uzbekistan  2014  female    75+ years            9      348465   
27817  Uzbekistan  2014    male  05-14 years           60     2762158   
27818  Uzbekistan  2014  female  05-14 years           44     2631600   
27819  Uzbekistan  2014  female  55-74 years           21     1438935   

       suicides/100k pop    country-year  HDI_for_year  gdp_for_year  \
0                   6.71     Albania1987           NaN    2156624900   
1                   5.19     Albania1987           NaN    2156624900   
2                   4.83     Albania1987           NaN    2156624900   
3                   4.59     Albania1987           NaN    2156624900   
4                   3.28     Albania1987           NaN    2156624900   
...                  ...             ...           ...           ...   
27815               2.96  Uzbekistan2014         0.675   63067077179   
27816               2.58  Uzbekistan2014         0.675   63067077179   
27817               2.17  Uzbekistan2014         0.675   63067077179   
27818               1.67  Uzbekistan2014         0.675   63067077179   
27819               1.46  Uzbekistan2014         0.675   63067077179   

       gdp_per_capita       generation  
0                 796     Generation X  
1                 796           Silent  
2                 796     Generation X  
3                 796  G.I. Generation  
4                 796          Boomers  
...               ...              ...  
27815            2309     Generation X  
27816            2309           Silent  
27817            2309     Generation Z  
27818            2309     Generation Z  
27819            2309          Boomers  

[27820 rows x 12 columns]

Convert into numerical variables

In [970]:
df['sex'] = df['sex'].map({'male': 1, 'female': 0})
age_mapping = {
    '05-14 years': 10,  # Approximate midpoint
    '15-24 years': 20,
    '25-34 years': 30,
    '35-54 years': 45,
    '55-74 years': 65,
    '75+ years': 85
}
df['age'] = df['age'].map(age_mapping)

generation_mapping = {
    'Generation Z': 6,
    'Millenials': 5,
    'Generation X': 4,
    'Boomers': 3,
    'Silent': 2,
    'G.I. Generation': 1
}
df['generation'] = df['generation'].map(generation_mapping)

df=df.drop(columns=['HDI_for_year', 'suicides_no', 'country-year','gdp_per_capita','population'])
df

country  year  sex  age  suicides/100k pop  gdp_for_year  generation
0         Albania  1987    1   20               6.71    2156624900           4
1         Albania  1987    1   45               5.19    2156624900           2
2         Albania  1987    0   20               4.83    2156624900           4
3         Albania  1987    1   85               4.59    2156624900           1
4         Albania  1987    1   30               3.28    2156624900           3
...           ...   ...  ...  ...                ...           ...         ...
27815  Uzbekistan  2014    0   45               2.96   63067077179           4
27816  Uzbekistan  2014    0   85               2.58   63067077179           2
27817  Uzbekistan  2014    1   10               2.17   63067077179           6
27818  Uzbekistan  2014    0   10               1.67   63067077179           6
27819  Uzbekistan  2014    0   65               1.46   63067077179           3

[27820 rows x 7 columns]

In [971]:
df = pd.get_dummies(df, 'country',drop_first=True)

X = df.drop(columns=['suicides/100k pop'])
y = df['suicides/100k pop']


model = LinearRegression()
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42,shuffle=True)

model.fit(X_train, y_train)

num_coefficients = len(model.coef_) + 1 # Add the intercept

print("Number of regression coefficients:", num_coefficients)



Number of regression coefficients: 106


The number of coefficients is 106.

In [972]:
y_pred = model.predict(X_test)

mae = mean_absolute_error(y_test, y_pred)
print("Mean Absolute Error (MAE):", mae)

Mean Absolute Error (MAE): 8.956041487694119


## 4. Use this new Q3. model to predict the target value for the people with age 20, male, and generation X. Report the prediction. What is the MAE error of this prediction? 

In [973]:
target_X = X[(X['sex'] == 1) & (X['age'] == 20) & (X['generation'] == 4)]
target_y = y[(X['sex'] == 1) & (X['age'] == 20) & (X['generation'] == 4)]
target_X

year  sex  age  gdp_for_year  generation  country_Antigua and Barbuda  \
0      1987    1   20    2156624900           4                        False   
13     1988    1   20    2126000000           4                        False   
28     1989    1   20    2335124988           4                        False   
37     1992    1   20     709452584           4                        False   
48     1993    1   20    1228071038           4                        False   
...     ...  ...  ...           ...         ...                          ...   
27632  1996    1   20   13948892216           4                        False   
27644  1997    1   20   14744603774           4                        False   
27656  1998    1   20   14988971211           4                        False   
27667  1999    1   20   17078465982           4                        False   
27679  2000    1   20   13760374488           4                        False   

       country_Argentina  country_Armenia  country_Aruba  country_Australia  \
0                  False            False          False              False   
13                 False            False          False              False   
28                 False            False          False              False   
37                 False            False          False              False   
48                 False            False          False              False   
...                  ...              ...            ...                ...   
27632              False            False          False              False   
27644              False            False          False              False   
27656              False            False          False              False   
27667              False            False          False              False   
27679              False            False          False              False   

       ...  country_Thailand  country_Trinidad and Tobago  country_Turkey  \
0      ...             False                        False           False   
13     ...             False                        False           False   
28     ...             False                        False           False   
37     ...             False                        False           False   
48     ...             False                        False           False   
...    ...               ...                          ...             ...   
27632  ...             False                        False           False   
27644  ...             False                        False           False   
27656  ...             False                        False           False   
27667  ...             False                        False           False   
27679  ...             False                        False           False   

       country_Turkmenistan  country_Ukraine  country_United Arab Emirates  \
0                     False            False                         False   
13                    False            False                         False   
28                    False            False                         False   
37                    False            False                         False   
48                    False            False                         False   
...                     ...              ...                           ...   
27632                 False            False                         False   
27644                 False            False                         False   
27656                 False            False                         False   
27667                 False            False                         False   
27679                 False            False                         False   

       country_United Kingdom  country_United States  country_Uruguay  \
0                       False                  False            False   
13                      False                  False            False   
28              

In [974]:
y_pred = model.predict(target_X)
print(f"Predicted suicide rate (suicides/100k pop): {np.mean(y_pred):.2f}")

mae = mean_absolute_error(target_y, y_pred)
print("Mean Absolute Error (MAE):", mae)

Predicted suicide rate (suicides/100k pop): 15.31
Mean Absolute Error (MAE): 6.169154376454643


## 5. Did you note any change in these two model performances?
Yes, the new model predict overall lower suicide rate and have smaller MAE, which suggests that the model have different coefficients to fit with these numerical features. The lower MAE suggests that the feature engineer improves the performance of this model. 

## 6. Use your Q3. model to predict the target value for age 33, male, and generation Alpha (i.e. the generation after generation Z); report the prediction

Let generation Alpha set to 7

In [975]:
X

year  sex  age  gdp_for_year  generation  country_Antigua and Barbuda  \
0      1987    1   20    2156624900           4                        False   
1      1987    1   45    2156624900           2                        False   
2      1987    0   20    2156624900           4                        False   
3      1987    1   85    2156624900           1                        False   
4      1987    1   30    2156624900           3                        False   
...     ...  ...  ...           ...         ...                          ...   
27815  2014    0   45   63067077179           4                        False   
27816  2014    0   85   63067077179           2                        False   
27817  2014    1   10   63067077179           6                        False   
27818  2014    0   10   63067077179           6                        False   
27819  2014    0   65   63067077179           3                        False   

       country_Argentina  country_Armenia  country_Aruba  country_Australia  \
0                  False            False          False              False   
1                  False            False          False              False   
2                  False            False          False              False   
3                  False            False          False              False   
4                  False            False          False              False   
...                  ...              ...            ...                ...   
27815              False            False          False              False   
27816              False            False          False              False   
27817              False            False          False              False   
27818              False            False          False              False   
27819              False            False          False              False   

       ...  country_Thailand  country_Trinidad and Tobago  country_Turkey  \
0      ...             False                        False           False   
1      ...             False                        False           False   
2      ...             False                        False           False   
3      ...             False                        False           False   
4      ...             False                        False           False   
...    ...               ...                          ...             ...   
27815  ...             False                        False           False   
27816  ...             False                        False           False   
27817  ...             False                        False           False   
27818  ...             False                        False           False   
27819  ...             False                        False           False   

       country_Turkmenistan  country_Ukraine  country_United Arab Emirates  \
0                     False            False                         False   
1                     False            False                         False   
2                     False            False                         False   
3                     False            False                         False   
4                     False            False                         False   
...                     ...              ...                           ...   
27815                 False            False                         False   
27816                 False            False                         False   
27817                 False            False                         False   
27818                 False            False                         False   
27819                 False            False                         False   

       country_United Kingdom  country_United States  country_Uruguay  \
0                       False                  False            False   
1                       False                  False            False   
2               

In [976]:
model = LinearRegression()

model.fit(X_train, y_train)

LinearRegression()

In [977]:
input_df=pd.DataFrame(0, index=[0],columns=X.columns)
input_data = {
    'sex': [1],  # Male
    'age': [33],
    'generation': [7]
}

input_df['sex'] = input_data['sex']
input_df['age'] = input_data['age']
input_df['generation'] = input_data['generation']
input_df[f'country_{most_common_country}'] = 1 
input_df['year'] = most_recent_year
input_df

year  sex  age  gdp_for_year  generation  country_Antigua and Barbuda  \
0  2016    1   33             0           7                            0   

   country_Argentina  country_Armenia  country_Aruba  country_Australia  ...  \
0                  0                0              0                  0  ...   

   country_Thailand  country_Trinidad and Tobago  country_Turkey  \
0                 0                            0               0   

   country_Turkmenistan  country_Ukraine  country_United Arab Emirates  \
0                     0                0                             0   

   country_United Kingdom  country_United States  country_Uruguay  \
0                       0                      0                0   

   country_Uzbekistan  
0                   0  

[1 rows x 105 columns]

In [978]:
prediction = model.predict(input_df)[0]
print(f"Predicted suicide rate: {prediction:.2f} per 100k population")

Predicted suicide rate: 26.84 per 100k population


## 7. Give one advantage when using regression (as opposed to classification with nominal features) in terms of independent variables.
The advantage is its flexibility in handling continuous data. While classification models typically deal with categorical outcomes, regression can find the linear or non-linear relationships between continuous independent variables and a dependent variable. It can quantitatively predict the result and the magnitude indicates the correlation. When there are some flexible changes in input data, such as "Generation Alpha"; new range of age, regression can still present relatively accurate result. However, classification may fail to predict. Also, the outcome for classification cannot suggest the strength of predictions.

## 8. Give one advantage when using regular numerical values rather than one-hot encoding for regression.
The advatage is that it can suggest the ordinal relationships and reduce the dimensionality. One-hot encoding separate the features into different variables but converting into numerical values preserve the ordinal relationships in the model such as age range and generation types. Also, it have fewer dimensionality considering the coefficients, leading to better performance. 

## 9. Now that you developed both a classifier (previously) and a regression model for the problem in this assignment, which method do you suggest to your machine learning model customer? Classifier or regression? Why?
### Classification is recommended

- For regression, an MAE of approximately 8.956 is quite high. This implies that the average prediction error is larger than the maximum value of the target range itself, indicating a significant model inaccuracy or inadequacy in capturing the patterns of the dataset.
- For classification model, the accuracy is 89.27%, demonstrating a high degree of accuracy in predicting the binary classes. Also, the metrix suggests that it is balanced across both target classes. This suggests that the model is not only accurate overall but also fair and effective across different scenarios within the dataset.

Though the suicides/100k pop is numerical values, considering the low performance of regression, classification is recommended. Meanwhile, classification model can extract the feature ranking, which can identify the important features contributing to higher suicides rate. The ability to explain outcomes as categories can simplify the processes and outcomes.